pkg/registry/scripts/main.js

```diff
@@ -146,38 +149,63 @@
                 connect();
             };
 
-            /*
-             * HACK: Because we don't have access to the information
-             * about the docker-registry Route, we cannot lookup its
-             * hostname information. So for now lets just look at an
-             * image stream.
-             */
-            var discoverHost = function() {
-                var host = null;
-                angular.forEach(select().kind("ImageStream"), function(stream) {
-                    var repo, status = stream.status || {};
-                    if (!host) {
-                        repo = status.dockerImageRepository || "";
-                        host = repo.split("/")[0];
-                    }
-                });
-                return host;
-            };
-
             /* When the loader changes digest */
             loader.listen(function() {
-                var host;
+                $rootScope.$applyAsync();
+            });
+        }
+    ])
+
+    .factory('registrySettings', [
+        '$q',
+        'KubeSettings',
+        'cockpitKubeDiscover',
+        function($q, KubeSettings, discover) {
+            return function registrySettings() {
+                var data = { };
 
-                if (discoverHost) {
-                    host = discoverHost();
-                    if (host) {
-                        $scope.settings.registry.host = host;
-                        discoverHost = null;
+                var one = discover().then(function(options) {
+                    data.password = null;
+
+                    /* See if we have a bearer token to use */
+                    var authorization, pos;
+                    if (options.headers) {
+                        authorization = (options.headers['Authorization'] || "").trim();
+                        if (authorization.toLowerCase().indexOf("bearer ") === 0)
+                            data.password = authorization.substr(7).trim();
+                    }
+                });
+
+                function processSettings(env) {
+                    /*
+                     * HACK: Because openshift doesn't give us the information we need
+                     * to look at where the registry is, we have to guess.
+                     *
+                     * We use an proper environment variable that comes in via KubeSettings.
+                     */
+
+                    var parser;
+                    var url = env["COCKPIT_KUBE_URL"];
```

Comment:
We're already passing this in for the oauth client object so this seems fine to me.